treeherder/log_parser/utils.py

```diff
@@ -1,66 +1,31 @@
-import logging
-
 import simplejson as json
-from requests.exceptions import HTTPError
-
-from treeherder.etl.artifact import (serialize_artifact_json_blobs,
-                                     store_job_artifacts)
-from treeherder.log_parser.artifactbuildercollection import (ArtifactBuilderCollection,
-                                                             LogSizeException)
-from treeherder.model.models import JobLog
-
-logger = logging.getLogger(__name__)
-
-
-def extract_text_log_artifacts(job_log):
-    """Generate a set of artifacts by parsing from the raw text log."""
-
-    # parse a log given its url
-    artifact_bc = ArtifactBuilderCollection(job_log.url)
-    artifact_bc.parse()
-
-    artifact_list = []
-    for name, artifact in artifact_bc.artifacts.items():
-        artifact_list.append({
-            "job_guid": job_log.job.guid,
-            "name": name,
-            "type": 'json',
-            "blob": json.dumps(artifact)
-        })
+from jsonschema import (ValidationError,
+                        validate)
 
-    return artifact_list
+with open('schemas/performance-artifact.json') as f:
+    PERFHERDER_SCHEMA = json.load(f)
 
 
-def post_log_artifacts(job_log):
-    """Post a list of artifacts to a job."""
-    logger.debug("Downloading/parsing log for log %s", job_log.id)
+def validate_perf_data(performance_data: dict):
+    validate(performance_data, PERFHERDER_SCHEMA)
+    expected_range = (45, PERFHERDER_SCHEMA["definitions"]["suite_schema"]["properties"]["extraOptions"]["items"]["maxLength"])
```

Comment:
This deep dictionary lookup could look better. I suggest move it up as a global module-level variable. Even use a small helper function for hide away that lookup.
Also, we should provide a unit test over the schema itself, to make sure no one is simply removing this expected field from the definition.

Reply:
Solved.